packages/table/preview/index.tsx

```diff
@@ -941,3 +953,13 @@ function getRandomInteger(min: number, max: number) {
 function contains(arr: any[], value: any) {
     return arr.indexOf(value) >= 0;
 }
+
+function localStorageGetOrDefault(stateKey: keyof IMutableTableState, defaultValue: any) {
+    const storedState = localStorage.getItem(stateKey);
+    if (storedState == null) {
+        // if we're not in local storage, populate local storage for next time
```

Comment:
Hmm, this is sneaky. And we don't _really_ need this helper, since it's only used once. Another approach could be to make the constructor a little busier but the code less surreptitious:

```tsx
// populate the state from local storage if the values are there;
// otherwise, use (and store) the defaults.
Object.keys(stateDefaults).forEach((stateKey: keyof IMutableTableState) => {
    let stateValue = localStorage.getItem(stateKey);

    if (stateValue == null) {
      // populate local storage for next time.
      stateValue = JSON.stringify(stateDefaults[stateKey]);
      localStorage.setItem(stateKey, stateValue);
    }

    this.state[stateKey] = JSON.parse(stateValue);
});
```